test/Microsoft.NET.Restore.Tests/GivenThatWeWantToUseFrameworkRoslyn.cs

```diff
@@ -46,6 +46,44 @@ public void It_restores_Microsoft_Net_Compilers_Toolset_Framework_when_requested
             }
         }
 
+        [Fact]
+        public void It_restores_Microsoft_Net_Compilers_Toolset_Framework_when_MSBuild_is_torn()
+        {
+            const string testProjectName = "NetCoreApp";
+            var project = new TestProject
+            {
+                Name = testProjectName,
+                TargetFrameworks = "net6.0",
+            };
+
+            // Add an explicit version for the test. This will normally come from the installer.
+            project.AdditionalProperties.Add("_NetFrameworkHostedCompilersVersion", "4.7.0-2.23260.7");
+            // simulate mismatched MSBuild versions
+            project.AdditionalProperties.Add("_IsDisjointMSBuildVersion", "true");
```

Comment:
Consider overriding `_BundledMSBuildVersionMajorMinor` instead of `_IsDisjointMSBuildVersion`, I think that would exercise more of the logic.

Reply:
Ah, and use a version that would guarantee a torn MSBuild?

Reply:
turns out that we can't use the `_BundledMSBuildVersionMajorMinor ` property because that property only exists at the installer-build-time when the bundled MSBuild props file is generated. I'm going to revert this change as a result.